Smt/Nat.lean

```diff
@@ -10,27 +10,39 @@ open Smt.Transformer
 /-- Removes casts of literals to `Nat` in `e`. For example, given
     `(app (app (app (OfNat.ofNat ..) ..) (LIT 0) ..) ..)`, this method removes
     all applications and returns just `(LIT 0)`. -/
-@[Smt] def markNatLiterals : Transformer
+@[Smt] def removeCasts : Transformer
```

Comment:
```suggestion
@[Smt] def removeOfNat : Transformer
```
I think "cast" is usually used to refer to typecasting by equality with `eq.rec`.